### PR TITLE
Get rid of profile references and use request.user everywhere

### LINKED ISSUE #3459: Completely get rid of profile references
Like `request.profile`.

### PR DESCRIPTION
**Pull request by julen**
I had to duplicate a _minimal_ part of `django.contrib.auth.middleware` but I think it's an acceptable trade-off. Now one can simply use `request.user` and forget about `request.profile` altogether.

Fixes #3459.

### CODE REVIEW

**Comment by phlax** [comment]
lgtm - havent done in clicky testing - nice one for cleaning this up!

**Comment by unho** [comment]
Spoke too soon on #4508.

File: pootle/middleware/auth.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# AUTHORS file for copyright and authorship information.
+
+"""
+Custom Pootle authentication middleware that takes care of returning Pootle's
+special `nobody` user instead of Django's own `AnonymousUser`.
+
+The code has been slightly adapted from django.contrib.auth.middleware.
+
+Note this customization would probably be unnecessary if there was a fix for
+https://code.djangoproject.com/ticket/20313
```
Comment (unho) [inline pootle/middleware/auth.py]: This ticket being fixed also will render unnecessary `User.get` so perhaps it is a good idea to put some fixme note there as well pointing to this ticket.

**Comment by unho** [comment]
I would also replace `profile` references in https://github.com/translate/pootle/blob/master/pootle/apps/pootle_profile/templatetags/profile_tags.py#L17-L18

Did some click testing and seems to work fine.

lgtm once nits addressed.

File: pootle/middleware/auth.py (hunk 1 above)
Comment (julen) [inline pootle/middleware/auth.py]: Thanks for the pointer @unho. In fact, I ended up removing the `User.get` method as it has a single and very specific use which is fine to do directly in the middleware, as opposed to the previous scenario where it was necessary to do it manually in view code. So fortunately there's no need to be concerned about it anymore :smile:

Comment (unho) [inline pootle/middleware/auth.py]: Nice!

**Pull request merged**